src/memkind_memtier.c

```diff
@@ -742,6 +768,10 @@ memtier_builder_construct_memtier_memory(struct memtier_builder *builder)
 
 MEMKIND_EXPORT void memtier_delete_memtier_memory(struct memtier_memory *memory)
 {
+    if (memory->get_kind == memtier_policy_data_movement_get_kind) {
```

Comment:
Looks like a hack - checking function pointer instead of some explicit enum... ok for POC, but maybe some comment might be a good idea?